Add default labels to solution operator deployment & pod

alexandre-allard opened the pull request:
**Component**: salt, solutions

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 

**Summary**:

**Acceptance criteria**: 


---

Closes: #2366

alexandre-allard committed:
salt: add labels to solution operator and ui objects

add all kubernetes default labels to the solution
operator and UI objects

Refs: #2366

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2366-add-solution-ver-to-op-and-ui` into
`development/2.4`
* [w/2.5/improvement/2366-add-solution-ver-to-op-and-ui](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/2366-add-solution-ver-to-op-and-ui) will be merged into `development/2.5` 
* [w/2.6/improvement/2366-add-solution-ver-to-op-and-ui](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/2366-add-solution-ver-to-op-and-ui) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

Ebaneck commented on salt/metalk8s/orchestrate/solutions/files/operator/deployment.yaml:
```diff
@@ -3,15 +3,29 @@ kind: Deployment
+    app.kubernetes.io/instance: {{ solution }}-operator
+    app.kubernetes.io/version: {{ version }}
+    app.kubernetes.io/component: operator
+    app.kubernetes.io/part-of: {{ solution }}
+    app.kubernetes.io/managed-by: salt
 spec:
   replicas: 1
   selector:
     matchLabels:
-      name: {{ solution }}-operator
+      app: {{ solution }}-operator
```
I think this change here will break the Solutions test.
It is related to this line and must be fixed
`        Then we have 1 running pod labeled 'name=example-solution-operator' in namespace 'example-environment'
`

Ebaneck commented on salt/metalk8s/orchestrate/solutions/files/operator/deployment.yaml:
```diff
@@ -38,4 +52,4 @@ spec:
           configMap:
-            name: {{ solution }}-operator-config
\ No newline at end of file
+            name: {{ solution }}-operator-config
```
What changed here??

alexandre-allard replied:
New line at the end of the file

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @ebaneck

alexandre-allard committed:
tests: adapt example solution operator test label

change from `name` to `app` to reflect the labels
changes

Refs: #2366

alexandre-allard commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @ebaneck







*The following options are set:* **approve**

gdemonet reviewed (changes requested):
Could you also add `heritage` (equivalent of `managed-by`, often used by Helm)? Basically, make sure that both Operator and UI deployments use the same labelling principles (not just the *Deployment*s, all objects can/should be labelled).

It could be a feature of our `metalk8s_kubernetes` module to enforce those labels are always set, like it does for automatically adding the `metalk8s.scality.com/version` label... WDYT @TeddyAndrieux @NicolasT?

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet

* @ebaneck







*The following options are set:* **approve**

alexandre-allard committed:
salt: add labels by default on every object

add `heritage` and `app.kubernetes.io/managed-by`
labels with `salt` as the value on each and every
objects we deploy using our Kubernetes module

bert-e commented:
# History mismatch



Merge commit #607f50daad21ed3b362d9d89e342cef3fdc50d60 on the integration branch
`w/2.5/improvement/2366-add-solution-ver-to-op-and-ui` is merging a branch which is neither the current
branch `improvement/2366-add-solution-ver-to-op-and-ui` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `improvement/2366-add-solution-ver-to-op-and-ui` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2366-add-solution-ver-to-op-and-ui` into
`development/2.4`
* [w/2.5/improvement/2366-add-solution-ver-to-op-and-ui](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/2366-add-solution-ver-to-op-and-ui) will be merged into `development/2.5` 
* [w/2.6/improvement/2366-add-solution-ver-to-op-and-ui](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/2366-add-solution-ver-to-op-and-ui) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet

* @ebaneck







*The following options are set:* **approve**

gdemonet commented on salt/_modules/metalk8s_kubernetes.py:
```diff
@@ -139,6 +139,9 @@ def method(manifest=None, name=None, kind=None, apiVersion=None,
             manifest.setdefault('metadata', {}).setdefault('labels', {})[
                 'metalk8s.scality.com/version'
             ] = match.group('version') if match else "unknown"
+            manifest['metadata']['labels']['app.kubernetes.io/managed-by'] = \
+                'salt'
+            manifest['metadata']['labels']['heritage'] = 'salt'
```
:+1:

TeddyAndrieux replied:
Right now for things like coredns we put "metalk8s" as heritage

Edit: And for ingress we use "metalk8s" as managed-by also .... anyway we should have the same thing everywhere
Heritage = metalk8s
Managed-by = salt ?

TeddyAndrieux replied:
Also I think we shouldn't override these value if they are already in the manifest (and indeed it's the same for version just above)

gdemonet reviewed (approved):
LGTM!

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @ebaneck







*The following options are set:* **approve**

alexandre-allard-scality commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye alexandre-allard-scality.
